lib/control-connection.js

```diff
@@ -415,6 +415,28 @@ class ControlConnection extends events.EventEmitter {
     }
   }
 
+  async _refreshControlConnection(hostIterator) {
+
+    if (this.options.sni) {
+      this.connection = this._borrowAConnection(hostIterator);
+    }
+    else {
+      try { this.connection = this._borrowAConnection(hostIterator); }
+      catch(err) {
+
+        /* NODEJS-632: refresh nodes before getting hosts for reconnect since some hostnames may have
+         * shifted during the flight. */
+        this.log("info", "ControlConnection could not reconnect using existing connections.  Refreshing contact points and retrying");
+        this._contactPoints.clear();
+        this._resolvedContactPoints.clear();
+        await Promise.all(this.options.contactPoints.map(name => this._parseContactPoint(name)));
```

Comment:
Does _parseContactPoint() stuff the result into _contactPoints?

Followup question, what happens if that fails? What's the end result in _contactPoints that we're going to use in the next line?

Reply:
Yes, _parseContactPoint() calls _addContactPoint in various ways which winds up populating _contactPoints.

We're mirroring the logic that takes place in initialization again... see https://github.com/datastax/nodejs-driver/blob/nodejs-632/lib/control-connection.js#L186 specifically.  If we successfully resolve a hostname all the resulting IPs are added, otherwise we store the hostname along with an indicator saying resolution failed... see https://github.com/datastax/nodejs-driver/blob/nodejs-632/lib/control-connection.js#L160-L171 for the details.